### ISSUE TITLE
Scouter errors in Cutting Edge

### ISSUE THREAD

**Issue by Alceris** [Putnam3145/SPARKING-Dwarf-Fortress-DBZ-mod#25]
...ds\Dwarf Fortress Z Cutting the Edge\hack\lua\dfhack.lua:454: ...the Edge/data/save/region1/raw/scripts/dragonball/ki.lua:154: ')' expected near ','
stack traceback:
    [C]: in function 'error'
    ...ds\Dwarf Fortress Z Cutting the Edge\hack\lua\dfhack.lua:454: in function 'run_script_with_env'
    ...ds\Dwarf Fortress Z Cutting the Edge\hack\lua\dfhack.lua:421: in function 'script_environment'
    ...ta/save/region1/raw/scripts/dragonball/dwarf_scouter.lua:5: in function 'getPowerLevel'
    ...ta/save/region1/raw/scripts/dragonball/dwarf_scouter.lua:164: in function <...ta/save/region1/raw/scripts/dragonball/dwarf_scouter.lua:158>
    [C]: in ?

Also, dragonball/scouter does nothing.

**Comment by Alceris** [comment]
I think the problem is that you put a comma where you meant to put a period in the ki.lua.

**Comment by lethosor** [comment]
That or the comma is inside a set of parentheses that it should be outside of.

**Comment by Alceris** [comment]
No, it was definitely a comma where there should've been a period, however, that's not the only problem, as, after fixing that, my max Ki is showing as "1.#INF".

**Comment by lethosor** [comment]
Okay, I didn't remember that there were about there were five open parentheses after `math.floor` - I was thinking the `,262519` could have been intended as another argument to a function, but it's unlikely that it was given that it's nested several levels in from a function call.
For future reference, the error is occurring [here](https://github.com/Putnam3145/SPARKING-Dwarf-Fortress-DBZ-mod/blob/cutting_edge/raw/scripts/dragonball/ki.lua#L154).

**Comment by Alceris** [comment]
Removing the period doesn't do anything, it still returns 1.#INF as my max Ki.

**Comment by Alceris** [comment]
I replaced the line with 
    return math.floor((boost+((2^((willpower+focus+endurance)/7228.262519))*2250)*multiplier)+.5)
and it worked, which doesn't make sense to me because, as far as I can tell, that should return exactly the same thing as the original line.

**Comment by Alceris** [comment]
I replaced the line with this
return math.floor((boost+((2^((willpower+focus+endurance)/7228.262519))****2250)****multiplier)+.5)
and it worked, which doesn't make any sense to me, because as far as I can tell, that should return exactly the same thing. Also Markdown is a massive pain in the ass when you're just trying to enter things that have asterisks or underscores or other things like that.

**Comment by lethosor** [comment]
That's not the same - you're adding `boost` after multiplying by `multiplier` instead of before. This would be the same thing:
```lua
return math.floor(((boost+(2^((willpower+focus+endurance)/7228.262519))*2250)*multiplier)+.5)
```

You can surround short code fragments with a single backtick, or longer ones with three.

**Comment by Alceris** [comment]
That line returns exactly the same number as mine does, and what the heck is a "backtick"?

**Comment by lethosor** [comment]
They don't return the same thing for me (with randomly-chosen numbers).

This is a backtick: `
(it's on the `~ key on US keyboards and in the page I linked to.)

**Comment by Alceris** [comment]
I still have no idea what a backtick is, and I'm testing both lines in-game and they're returning the same numbers for max Ki.

**Comment by lethosor** [comment]
The line in your comment and the line in mine? They won't always return the same thing, but it's possible that they will for certain parameters.

https://guides.github.com/features/mastering-markdown/#examples - click "Code"

**Comment by Alceris** [comment]
Again, I said I know what a backtick is now, and I tested both lines both in normal and Super Saiyan forms, they return the exact same number.

**Comment by lethosor** [comment]
I don't know what the range of the parameters is, but the two lines seem to return the same value for parameters between 0 and 1, but not for 0 and 1000. Running this in a Lua interpreter prints two different numbers, unless `* 1000` is removed:

```lua
willpower = math.random() * 1000
focus = math.random() * 1000
endurance = math.random() * 1000
boost = math.random() * 1000
multiplier = math.random() * 1000

print(math.floor((boost+((2^((willpower+focus+endurance)/7228.262519))*2250)*multiplier)+.5))
print(math.floor(((boost+(2^((willpower+focus+endurance)/7228.262519))*2250)*multiplier)+.5))
```

Also, you edited your comment. GitHub will show new comments as they are posted (without refreshing), but won't show edits, so this is what I was replying to - sorry about the confusion:

> I still have no idea what a backtick is, and I'm testing both lines in-game and they're returning the same numbers for max Ki.

**Comment by Alceris** [comment]
Well, we appear to have solved max Ki not displaying correctly, however now there's the problem of Ki apparently not providing nearly as much of a boost in attack power as it should. Also, the slowing effect from Ki differences still doesn't work in Adventurer Mode.

**Comment by lethosor** [comment]
Looks like this was fixed in 1d065160235e17793385c891473802d7dd144de2
